grommet/grommet
# Ensure TextInput respects defaultSuggestion prop when onChange is triggered

#### What does this PR do?

Ensures that TextInput respects the `defaultSuggestion` prop when the input's `onChange` is triggered. The `onChange` method was setting `activeSelectionIndex` to -1 when it should have been using `resetSelectionIndex`. This caused the `defaultSuggestion` prop not to be respected if the value of the text field changed.

#### Where should the reviewer start?

The easiest thing to do is probably to revert the one-line change to `TextInput.js` and then run tests, and confirm you get a failure.

#### What testing has been done on this PR?

This was discovered when attempting to integrate the `defaultSuggestion` prop into the app discussed in #4781 . I diagnosed the issue, confirmed that there was a gap in the test coverage, and wrote a test to cover the gap. Then I made the change to resolve the issue.

#### How should this be manually tested?

If you open the existing `Default suggestion` storybook, you can verify that with this change, typing in the text input does not affect the highlighted default suggestion. However, if you revert the change in `TextInput.js`, you will see that entering input into the text input will cause the highlight over the default suggestion to be cleared.

#### Any background context you want to provide?

The user-requested goal here is to be able to auto-highlight the first suggestion in the list at all times, so that the user can type a partial match and then hit Enter to select the first suggestion without having to use the mouse or arrow keys to explicitly select it.

#### What are the relevant issues?

I have not opened an issue for this because it was a simple fix and it was quicker to just open the PR directly, but I am happy to open one if that would be helpful.

#### Screenshots (if appropriate)

N/A

#### Do the grommet docs need to be updated?

No.

#### Should this PR be mentioned in the release notes?

Yes, this is a bugfix.

#### Is this change backwards compatible or is it a breaking change?

Backwards compatible.
